src/main/java/org/verdictdb/coordinator/VerdictResultStream.java

```diff
@@ -16,47 +16,25 @@
 
 package org.verdictdb.coordinator;
 
-import org.verdictdb.connection.DbmsQueryResult;
-import org.verdictdb.core.resulthandler.ExecutionResultReader;
-
 import java.util.Iterator;
 
-public class VerdictResultStream
+public abstract class VerdictResultStream
     implements Iterable<VerdictSingleResult>, Iterator<VerdictSingleResult> {
 
-  ExecutionResultReader reader;
-
-  ExecutionContext execContext;
-
-  public VerdictResultStream(ExecutionResultReader reader, ExecutionContext execContext) {
-    this.reader = reader;
-    this.execContext = execContext;
-  }
-
   // TODO
-  public VerdictResultStream create(VerdictSingleResult singleResult) {
-    return null;
-  }
+  public abstract VerdictResultStream create(VerdictSingleResult singleResult);
```

Comment:
Please change this to a factory method that basically constructs an appropriate regular class and returns it.

Reply:
Please add a similar factory method for the ExecutionResultReader.

Reply:
You mean create a factory class like queryExecutionPlanFactory that implement create() method?

Reply:
Please ignore my above comment, and just convert this class to an interface. Since the interface cannot include any method body in JDK7, my comment is not applicable.